packages/components/menu/src/menu-item.ts

```diff
@@ -0,0 +1,24 @@
+import { buildProp, definePropType } from '@element-plus/utils/props'
+import { isString } from '@element-plus/utils/util'
+
+import type { MenuItemRegistered } from './types'
+import type { ExtractPropTypes } from 'vue'
+import type { RouteLocationRaw } from 'vue-router'
+
+export const menuItemProps = {
+  index: {
+    type: String,
+    default: null,
+  },
+  route: buildProp({
+    type: definePropType<RouteLocationRaw>([String, Object]),
+  }),
+  disabled: Boolean,
+} as const
+export type MenuItemProps = ExtractPropTypes<typeof menuItemProps>
+
+export const menuItemEmits = {
+  click: (item: MenuItemRegistered) =>
+    isString(item.index) && Array.isArray(item.indexPath),
```

Comment:
Do we really need to implement the actual method here? Since I thought this should only be available for marking the type.